src/install.ts

```diff
@@ -1,55 +1,66 @@
 import VueClass from 'vue'
 
-export interface IMobxMethods {
-  reaction: any
+export type Disposer = () => void
+export type Getter = () => any
+export type Setter = (value: any) => void
+export type FromMobxEntries = {
+  key: string,
+  get: Getter,
+  set?: Setter
 }
+export type VueComputed = Getter | { get: Getter, set: Setter }
 
-export type Disposer = () => void
+export interface IMobxMethods {
+  reaction: (getter: Getter, reaction: (value: any) => void, any) => Disposer
+}
 
 declare module 'vue/types/options' {
   interface ComponentOptions<V extends VueClass> {
     fromMobx?: { [key: string]: (this: V) => any }
   }
 }
 
+const movueDisposersKey = '__movueDisposers__'
+
 export default function install(Vue: typeof VueClass, mobxMethods: IMobxMethods) {
 
   const defineReactive = (Vue as any).util.defineReactive
 
   function beforeCreate(this: VueClass) {
     const vm = this
-    getFromStoreEntries(vm).forEach(({ key, compute }) => {
-      defineReactive(vm, key, null, null, true)
-    })
+
+    vm.$options.computed = getFromStoreEntries(vm).reduce(
+      (computed, { key, set }) => {
+        const reactivePropertyKey = getReactivePropertyKey(key)
+        defineReactive(vm, reactivePropertyKey, null, null, true)
+
+        return Object.assign(
+          computed,
+          { [key]: createComputedField(vm, reactivePropertyKey, set) }
+        )
+      },
+      vm.$options.computed || {}
+    )
   }
 
   function created(this: VueClass) {
     const vm = this
-    const entries = getFromStoreEntries(vm)
-    if (entries.length <= 0) {
-      return
-    }
-
-    const disposers: Disposer[] = vm['__movueDisposers__'] = []
 
-    entries.forEach(({ key, compute }) => {
-      disposers.push(mobxMethods.reaction(
-        () => compute.apply(vm),
-        val => {
-          vm[key] = val
-        },
-        true
-      ))
+    const disposers: Disposer[] = getFromStoreEntries(vm).map(({ key, get }) => {
+      const reactivePropKey = getReactivePropertyKey(key)
+      const updateReactiveProperty = value => { vm[reactivePropKey] = value }
+      return mobxMethods.reaction(() => get.call(vm), updateReactiveProperty, true)
     })
+
+    vm[movueDisposersKey] = disposers
```

Comment:
Another thing I care about here: when there is no `fromMobx` content in the component definition, it will be better for us not to set the `__movueDisposers__ ` field of the component. That makes normal components cleaner.

Reply:
I agree. I will refactor that code.